editoast/src/views/timetable/train_schedule.rs

```diff
@@ -590,6 +609,309 @@ async fn occupancy_blocks(
     Ok(Json(occupancy_blocks_result))
 }
 
+#[derive(Debug, Default, Clone, Serialize, Deserialize, ToSchema)]
+struct TrackOccupancyForm {
+    train_schedule_ids: Vec<i64>,
+    op_id: String,
+    infra_id: i64,
+    electrical_profile_set_id: Option<i64>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
+struct TrackOccupancy {
+    time_begin: DateTime<Utc>,
+    #[schema(value_type = chrono::Duration, example = "PT5M")]
+    duration: PositiveDuration,
+    train_schedule_id: i64,
+}
+
+#[utoipa::path(
+    post, path = "",
+    tag = "train_schedule",
+    request_body = inline(TrackOccupancyForm),
+    responses(
+        (status = 200, description = "Track section occupancy periods for a set of train schedules", body = inline(HashMap<String, Vec<TrackOccupancy>>)),
+    ),
+)]
```

Comment:
The endpoint should be document: **not just the API but rather what it actually does**